prov/efa/test/efa_unit_test_mr.c

```diff
@@ -3,33 +3,150 @@
 
 #include "efa_unit_tests.h"
 
-void test_efa_mr_reg_counters(struct efa_resource **state)
+static void test_efa_mr_counters(struct efa_domain *efa_domain, struct fid_mr *mr, int mr_reg_count, int mr_reg_size, bool gdrcopy_flag)
```

Comment:
It's not testing "counters" only any more. we need a better name for this function, e.g. `test_efa_mr_impl`